src/js/components/FragmentList/fragmentList.jsx

```diff
@@ -88,41 +54,12 @@ const FragmentList = () => {
   const [fragments, setFragments] = useState(sortedFragmentsByTime);
 
   function sortByStatus() {
-    function getStatusSortWeight(status) {
-      let sortWeight = 0;
-
-      switch (status) {
-        case 'error':
-          sortWeight = 0;
-          break;
-        case 'other':
-          sortWeight = 1;
-          break;
-        case 'missing':
-          sortWeight = 1;
-          break;
-        case 'success':
-          sortWeight = 2;
-          break;
-        case 'unprocessed':
-          sortWeight = 3;
-          break;
-        default: sortWeight = 4;
-          break;
-      }
-      return sortWeight;
-    }
-
-    const sortedFragments = fragments.concat().sort((a, b) => {
-      const { status: statusA } = a.props;
-      const { status: statusB } = b.props;
-      const sortWeightA = getStatusSortWeight(statusA);
-      const sortWeightB = getStatusSortWeight(statusB);
-
-      if (sortWeightA < sortWeightB) return -1;
-      if (sortWeightA > sortWeightB) return 1;
-      return 0;
-    });
+    const sortOrder = ['success', 'other', 'missing', 'unprocessed', 'error'];
+    const ordering = {};
+    sortOrder.forEach((element, index) => { ordering[element] = index; });
```

Comment:
I would consider using reduce pattern here,
also sortById and sortByType looks fairly similar, how about creating higher order function that would receive sort comparator?